src/rebar3_typer_prv.erl

```diff
@@ -10,11 +10,53 @@
               {rebar_state, command_parsed_args, 1}]).
 
 -define(PROVIDER, typer).
--define(OPTS,
-        [{boolean, $b, "boolean", {boolean, false}, "Boolean example"},
-         {string, $s, "rebar-config", {string, "some.txt"}, "String example"},
-         {atom, $a, "ignore", atom, "Atom example"},
-         {atom_none, $n, "atom-none", {atom, none}, "Atom example, with default"}]).
+-define(OPTS, %% TODO: consider adding shorthand version to some (or all) options, even if it doesn't exist on TypEr itself
+        %% list of {Name,  ShortOpt,  LongOpt,    ArgSpec,   HelpMsg}
+        [{recursive,
+          undefined,
+          "recursive",
+          {boolean, false},
```

Comment:
I think the default should be `true` here. I know that it's `false` in typer… but that's because you have to specify the files to type… Here you don't. As a matter of fact, what if we just remove this option for now and use whatever suits us best internally?

Reply:
I don't think wrapping around TypEr whilst removing options is something good. If someone using this wants to force the tool not to be recursive, we should not disregard their opinion and have the tool run as we want.

On the matter of the default value for this option, I even misunderstood it completely. After looking at TypEr's code I can say this is a list of directories, so I will be updating this accordingly.

Reply:
Well… some options are in typer only because typer doesn't know about _default project folder structure_ like rebar3 does. In general, when building a rebar3 plugin… you don't want your users to be forced to specify _where to run the plugin_ (I.e. on which files)… rebar3 should know that already.